network/local.go

```diff
@@ -277,27 +277,47 @@ func (lc *LocalConn) Type() ConnType {
 // All operations are thread-safe.
 // The messages are marshalled and stored in the queue as a slice of bytes.
 type connQueue struct {
-	*sync.Cond
-	queue  [][]byte
-	closed bool
+	incomingQueue chan []byte
+	outgoingQueue chan []byte
+	closeCh       chan bool
+	wg            sync.WaitGroup
```

Comment:
I'll try and see if it works and is not ugly. Because if it's the word Collection that bothers you, think of this routine of a collection of size 1. The syntax of sync.WaitGroup is much clearer `Add(), Done(), Wait()` and it's really for:
> A WaitGroup waits for a collection of goroutines to finish

I want my listening go routine to finish.